libafl/src/mutators/mutations.rs

```diff
@@ -64,9 +64,13 @@ pub fn buffer_set<T: Clone>(data: &mut [T], from: usize, len: usize, val: T) {
 #[inline]
 pub fn rand_range<S: HasRand>(state: &mut S, upper: usize, max_len: usize) -> Range<usize> {
     let len = state.rand_mut().next() as usize % max_len + 1;
-    let offset2 = state.rand_mut().next() as usize % (upper + len);
```

Comment:
I think this should use `rand_between`, else it'll be biased, right?

Reply:
I don't know the mathematical explanation on why this is biased or un-biased
cc @addisoncrump 

but there's several tests down here
https://github.com/AFLplusplus/LibAFL/blob/mutator_fix/libafl/src/mutators/mutations.rs#L1455
to ensure that the resulting mutation is not biased. so I guess it's fine as long as it passes all the test

Reply:
rand with modulo will always be biased. Why not `rand_between`? does it have any speed inpact?

Reply:
I tried rand.between()

it has speed impact
```
delete 31ns
expand 43ns
insert 38ns
rand insert 37ns
set 31ns
rand set 30ns
copy 37ns
insert copy 62ns
swap 75ns
crossover insert 94ns
crossover replace 69ns
```
so 547ns in total

previously https://github.com/AFLplusplus/LibAFL/pull/1138#issuecomment-1465241261 
this has 504ns in total.
so I'd keep this below()

Reply:
it's not `below` here, it's `%` (?)
Ultimately it's a trade-off, speed or correct randomness, right? I'm not sure if saving 40ns is worth it

Reply:
What does thumbs up mean? :D you're gonna change it or no? :P

Reply:
Apart from this tiny nit, we can merge this from my PoV